src/state/machines/guards/buy.ts

```diff
@@ -23,13 +23,18 @@ export const BuyMachineGuards = {
           ?.hand.filter(({ uid }) => turn.playedCards.includes(uid))
           .filter(({ type }) => type === "plant") as PlantCard[]) ?? [];
 
-      return playedPlants.filter(({ biomes }) => intersection(biomes, requiredBiomes).length > 0).length >= 2;
+      return playedPlants.filter(({ biomes }) => requiredBiomes.every((b) => biomes.includes(b))).length >= 2;
```

Comment:
I think the logic here is still incorrect. In this case the plants would need all of the biomes that the animal has, which is not the case. If AT LEAST one of the animal biomes are shared by two plants it can be bought

Reply:
![image](https://github.com/user-attachments/assets/f84ce8b6-9c26-4143-9e9d-8dcb7ae793c5)

Here is an example, the other animal with P should also be available